src/lib/tls/tls13_pqc/hybrid_public_key.cpp

```diff
@@ -0,0 +1,373 @@
+/**
+* Composite key pair that exposes the Public/Private key API but combines
+* multiple key agreement schemes into a hybrid algorithm.
+*
+* (C) 2023 Jack Lloyd
+*     2023 Fabian Albert, René Meusel - Rohde & Schwarz Cybersecurity
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
```

Comment:
General comment on this: it seems like the implementation is rather complicated by supporting an arbitrary number of keys, instead of exactly 2. The draft does allow it but it seems implausible to me that anyone would ever use it.

Reply:
Mhm, that's the conclusion we drew after finishing the general implementation. Perhaps even more simplification would be possible by supporting exactly one KEM and one KEX algorithm. Right now this class supports an arbitrary combination of 1..n such algorithms.

On the plus side: If one instantiates this as a wrapper around a single KEX, it acts as a KEM adapter. That may or may not be a useful bit.